genisys/modules/preseed.py

```diff
@@ -33,16 +33,18 @@ def generate(self: Self) -> str:
 
                 # Read SSH key files and store their contents as strings
                 ssh_keys_contents = []
-                ssh_keys_dir = Path(self.config["users"].get("ssh-keys-dir", "../ssh_keys"))
-                for ssh_key_file in self.config["users"].get("ssh-keys", []):
-                    ssh_key_path = ssh_keys_dir / ssh_key_file
-                    with open(ssh_key_path, 'r') as f:
-                        ssh_keys_contents.append(f.read())
+                for ssh_key_path_str in self.config["users"].get("ssh-keys", []):
+                    ssh_key_path = Path(ssh_key_path_str)  # Directly use the provided absolute path
+                    try:
+                        with open(ssh_key_path, 'r') as f:
+                            ssh_keys_contents.append(f.read())
+                    except FileNotFoundError:
+                        print(f"Warning: SSH key file {ssh_key_path} not found.")
 
                 # Determine SSL certificate path
-                ssl_cert_path = Path(self.config["network"].get("server", {}).get("ssl", {}).get("cert",
-                                                                                                 CERTIFICATE_STORE_PATH / "cert.pem"))
+                ssl_cert_path = Path(tls.get_keychain(self.config["network"]["server"]["ssl"])["certfile"])
```

Comment:
Be sure to only do anything related to SSL iff the 'ssl' key is in the config. If the 'ssl' key is missing the server runs in HTTP mode. I think the easiest way for you to accomplish this, based on the preseed template, is to specify something like the following
```python
ssl_cert_content = None
if 'ssl' in self.config["network"].get("server", {}):
    # the rest of the ssl logic
# the rest of the generate function
```

Reply:
Changed